src/main/java/com/onarandombox/MultiverseCore/commands/ImportCommand.java

```diff
@@ -0,0 +1,88 @@
+import com.onarandombox.MultiverseCore.MultiverseCore;
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandFlag;
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandFlagGroup;
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandValueFlag;
+import com.onarandombox.MultiverseCore.commandtools.flags.ParsedCommandFlags;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.World;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+@CommandAlias("mv")
+public class ImportCommand extends MultiverseCoreCommand {
+    public ImportCommand(@NotNull MultiverseCore plugin) {
+        super(plugin);
+
+        registerFlagGroup(CommandFlagGroup.builder("mvimport")
+                .add(CommandValueFlag.builder("--generator", String.class)
+                        .addAlias("-g")
+                        .completion(() -> Arrays.stream(Bukkit.getServer().getPluginManager().getPlugins())
+                                .filter(Plugin::isEnabled)
+                                .filter(genplugin -> this.plugin.getUnsafeCallWrapper().wrap(
+                                        () -> genplugin.getDefaultWorldGenerator("world", ""),
+                                        genplugin.getName(),
+                                        "Get generator"
+                                ) != null)
+                                .map(genplugin -> genplugin.getDescription().getName())
+                                .collect(Collectors.toList()))
+                        .build())
+                .add(CommandFlag.builder("--adjust-spawn")
+                        .addAlias("-a")
+                        .build())
+                .build());
+    }
+
+    @Subcommand("import")
+    @CommandPermission("multiverse.core.import")
+    @CommandCompletion("@mvworlds:scope=potential  @flags:groupName=mvimport")
+    @Syntax("<name> <env> --generator [generator[:id]] --adjust-spawn")
+    @Description("Imports a existing world folder.")
+    public void onImportCommand(BukkitCommandIssuer issuer,
+
+                                @Conditions("validWorldName:scope=new")
+                                @Syntax("<name>")
+                                @Description("Folder name of the world.")
+                                String worldName,
+
+                                @Syntax("<env>")
+                                @Description("The world's environment. See: /mv env")
+                                World.Environment environment,
+
+                                @Optional
+                                @Syntax("--generator [generator[:id]] --adjust-spawn")
+                                @Description("Other world settings. See: http://gg.gg/nn8c2")
+                                String[] flags) {
+
+        ParsedCommandFlags parsedFlags = parseFlags(flags);
+
+        issuer.sendMessage(String.format("Starting import of world '%s'...", worldName));
```

Comment:
No need to say `world <worldName>`, would be better as
```java
issuer.sendMessage(ChatColor.GREEN + "Starting import of " + ChatColor.GOLD + worldName);
```

Reply:
hmm yea, tbh we need to standardise our output formatting colour and stuff

Reply:
I have given thought to that problem before but had trouble coming up with a good solution. It's hard to come up with a concise list of "categories" to assign colors to.

Reply:
yea, i see other plugins most colours are hard coded also

Reply:
Currently not too concerned about how fancy the output is, will be moved to locales anyways